docs/site/Loopback-component-authentication.md

```diff
@@ -0,0 +1,600 @@
+- The binding key `AuthenticationBindings.METADATA.key` is bound to
+  `AuthMetadataProvider` which returns authentication decorator metadata of type
+  `AuthenticationMetadata`
+
+The purpose of these providers and the values they return will be explained in
+the sections below.
+
+## Authentication Strategies
+
+Support for multiple authentication strategies is possible with a **common**
+authentication strategy interface, and an **extensionPoint/extensions** pattern
+used to **register** and **discover** authentication strategies.
+
+### AuthenticationStrategy Interface
+
+The `AuthenticationComponent` declares a common authentication strategy
+interface named `AuthenticationStrategy`.
+
+```ts
+export interface AuthenticationStrategy {
+  /**
+   * The 'name' property is a unique identifier for the
+   * authentication strategy ( for example : 'basic', 'jwt', etc)
+   */
+  name: string;
+
+  /**
+   * The 'authenticate' method takes in a given request and returns a user profile
+   * which is an instance of 'UserProfile'.
+   * (A user profile is a minimal subset of a user object)
+   * If the user credentials are valid, this method should return a 'UserProfile' instance.
+   * If the user credentials are invalid, this method should throw an error
+   * If the user credentials are missing, this method should throw an error, or return 'undefined'
+   * and let the authentication action deal with it.
+   *
+   * @param request
+   */
+  authenticate(request: Request): Promise<UserProfile | undefined>;
+}
+```
+
+Developers that wish to create a custom authentication strategy must implement
+this interface. The custom authentication strategy must have a **unique** `name`
+and have an `authenticate` function which takes in a request and returns the
+user profile of an authenticated user.
+
+Here is an example of a basic authentication strategy named
+`BasicAuthenticationStrategy` in `basic-strategy.ts`:
+
+```ts
+export interface Credentials {
+  username: string;
+  password: string;
+}
+
+export class BasicAuthenticationStrategy implements AuthenticationStrategy {
+  name: string = 'basic';
+
+  constructor(
+    @inject(UserServiceBindings.USER_SERVICE)
+    private userService: UserService,
+  ) {}
+
+  async authenticate(request: Request): Promise<UserProfile | undefined> {
+    const credentials: Credentials = this.extractCredentials(request);
+    const user = await this.userService.verifyCredentials(credentials);
+    const userProfile = this.userService.convertToUserProfile(user);
+
+    return userProfile;
+  }
+
+  extractCredentials(request: Request): Credentials {
+    let creds: Credentials;
+
+    /**
+     * Code to extract the 'basic' user credentials from the Authorization header
+     */
+
+    return creds;
+  }
+}
+```
+
+As you can see in the example, an authentication strategy can inject custom
+services to help it accomplish certain tasks. A complete example of a **basic**
+authentication strategy can be found
```

Comment:
Suggested change (see if it makes sense to you):
See the complete example for [basic authentication strategy](link1) and [jwt authentication strategy](link2)